ddtrace/debugging/_capture/snapshot.py

```diff
@@ -32,21 +36,26 @@ def _capture_context(
     limits=CaptureLimits(),  # type: CaptureLimits
 ):
     # type: (...) -> Dict[str, Any]
-    return {
-        "arguments": {
-            n: utils.capture_value(v, limits.max_level, limits.max_len, limits.max_size, limits.max_fields)
-            for n, v in arguments
-        }
-        if arguments is not None
-        else {},
-        "locals": {
-            n: utils.capture_value(v, limits.max_level, limits.max_len, limits.max_size, limits.max_fields)
-            for n, v in _locals
+    with HourGlass(duration=CAPTURE_TIME_BUDGET) as hg:
+
+        def timeout(_):
+            return not hg.trickling()
+
+        return {
+            "arguments": {
+                n: utils.capture_value(v, limits.max_level, limits.max_len, limits.max_size, limits.max_fields, timeout)
+                for n, v in arguments
+            }
+            if arguments is not None
+            else {},
+            "locals": {
+                n: utils.capture_value(v, limits.max_level, limits.max_len, limits.max_size, limits.max_fields, timeout)
+                for n, v in _locals
+            }
+            if _locals is not None
+            else {},
+            "throwable": utils.capture_exc_info(throwable),
```

Comment:
Good question. I'd expect exception arguments to be simple objects, so serialisation shouldn't take long. Perhaps still worth trying to get more info about exceptions so best not to timeout on this?

Reply:
I agree. if we got an exception we better collect more data. but there are few points I believe we need to think about:
1. timeout is to prevent the program from getting stuck - so I don't see why capture exception doesn't need this protection. I'm open in increasing the timeout if you think it is too restrictive, but 200ms is long enough right?
2. if we believe we want to capture more data. why are using serialize(v) with depth=2? should we capture those as tree so the user can inspect those?